datalake/src/main/java/com/sequenceiq/datalake/service/sdx/database/DatabaseParameterFallbackUtil.java

```diff
@@ -0,0 +1,84 @@
+public class DatabaseParameterFallbackUtil {
+    private DatabaseParameterFallbackUtil() {
+    }
+
+    public static SdxDatabase setupDatabaseInitParams(SdxCluster sdxCluster, SdxDatabaseAvailabilityType databaseAvailabilityType, String dbEngineVersion) {
+        SdxDatabase sdxDatabase = new SdxDatabase();
+        sdxDatabase.setDatabaseAvailabilityType(databaseAvailabilityType);
+        sdxDatabase.setDatabaseEngineVersion(dbEngineVersion);
+        // TODO only for backward compatibilty, can be removed in CB-21369
+        sdxCluster.setDatabaseAvailabilityType(databaseAvailabilityType);
+        sdxCluster.setDatabaseEngineVersion(dbEngineVersion);
+        return sdxDatabase;
+    }
+
+    public static void setDatabaseCrn(SdxCluster sdxCluster, String databaseCrn) {
+        createSdxDatabaseIfMissing(sdxCluster);
+        sdxCluster.getSdxDatabase().setDatabaseCrn(databaseCrn);
+        // TODO only for backward compatibilty, can be removed in CB-21369
+        sdxCluster.setDatabaseCrn(databaseCrn);
+    }
+
+    public static String getDatabaseCrn(SdxDatabase sdxDatabase, String fallbackDatabaseCrn) {
+        if (sdxDatabase != null) {
+            return sdxDatabase.getDatabaseCrn();
+        } else {
+            return fallbackDatabaseCrn;
+        }
+    }
+
+    public static String getDatabaseEngineVersion(SdxDatabase sdxDatabase, String fallbackDatabaseEngineVersion) {
+        if (sdxDatabase != null) {
+            return sdxDatabase.getDatabaseEngineVersion();
+        } else {
+            return fallbackDatabaseEngineVersion;
+        }
+    }
+
+    public static boolean isCreateDatabase(SdxDatabase sdxDatabase, boolean fallbackCreateDatabase) {
+        if (sdxDatabase != null) {
+            return sdxDatabase.isCreateDatabase();
+        } else {
+            return fallbackCreateDatabase;
+        }
+    }
+
+    public static SdxDatabaseAvailabilityType getDatabaseAvailabilityType(SdxDatabase sdxDatabase, SdxDatabaseAvailabilityType fallbackDatabaseAvailabilityType,
+            boolean fallbackCreateDatabase) {
+        if (sdxDatabase != null) {
+            return getDatabaseAvailabilityType(sdxDatabase.getDatabaseAvailabilityType(), sdxDatabase.isCreateDatabase());
+        } else {
+            return getDatabaseAvailabilityType(fallbackDatabaseAvailabilityType, fallbackCreateDatabase);
+        }
+    }
+
+    public static SdxDatabaseAvailabilityType getDatabaseAvailabilityType(SdxDatabaseAvailabilityType databaseAvailabilityType, boolean createDatabase) {
+        if (databaseAvailabilityType != null) {
+            return databaseAvailabilityType;
+        } else {
+            if (createDatabase) {
+                return SdxDatabaseAvailabilityType.HA;
+            } else {
+                return SdxDatabaseAvailabilityType.NONE;
+            }
+        }
+    }
+
+    private static void createSdxDatabaseIfMissing(SdxCluster sdxCluster) {
```

Comment:
It can happen if cloudbreak services upgrade is happening during sdx cluster creation and the database creation flow is continued in an sdx service with a new version. This part can be removed in later versions.